File: src/main/java/src/syntax/annotations/example_6/AnnotationsMain.java
package src.syntax.annotations.example_6;

// import syntax.annotations.example_6.SuperComputer;

import syntax.annotations.example_6.MainKt;

public class AnnotationsMain {
    public static void main(String[] args) {

        /*
         * Object property
         */
        // #1 property inside the object without annotation
        // int answer = SuperComputer.INSTANCE.getAnswer();

        // #2 property inside the object with annotation '@JvmStatic'
        // int answer = SuperComputer.getAnswer();

        // #3 property inside the object with annotation '@JvmField'
        // int answer = SuperComputer.answer;

        // #4 property inside the object without annotation but with
        // 'const' keyword.
        // int answer = SuperComputer.answer;


        /*
         * Top level property
         */
        // #5 top level property without annotations
        // int answer = MainKt.getAnswer();

        // #6 top level property with annotation '@JvmField'
        // int answer = MainKt.answer;

        // #7 top level property without annotations but with
        // 'const' keyword.
        int answer = MainKt.answer;     // Value can be changed.
        answer = 19;

        System.out.println(
                "answer: " +answer
        );
    }
}

File: src/syntax/annotations/example3/AnnotationsMain.kt
package syntax.annotations.example3

fun main() {

}

// #1
//object SuperComputer {
//    val answer = 42
//}

// #2
//object SuperComputer {
//    @JvmStatic
//    val answer = 42
//}

// Exposing the field for all types of data
// #3
//object SuperComputer {
//    @JvmField
//    val answer = 42
//}

// Exposing the field for primitive and String type data
// #4
object SuperComputer {
    const val answer = 42
}

/**
 * Top level property
 */
// #5
// val answer = 42

// #6
//@JvmField
//val answer = 42

// #7
const val answer = 42